modules/apps/roles/roles-admin-web/src/main/resources/META-INF/resources/edit_role_assignments.jsp

```diff
@@ -118,105 +150,7 @@ renderResponse.setTitle(role.getTitle(locale));
-	<portlet:param name="mvcPath" value="/edit_role_assignments.jsp" />
-	<portlet:param name="tabs1" value="assignees" />
-</portlet:actionURL>
-
-<aui:script sandbox="<%= true %>">
-	var form = document.<portlet:namespace />fm;
-
-	var addAssignees = function (event) {
-		Liferay.Util.openSelectionModal({
-			multiple: true,
-			onSelect: function (selectedItem) {
-				if (selectedItem) {
-					var assignmentsRedirect = Liferay.Util.PortletURL.createPortletURL(
-						'<%= portletURL.toString() %>',
-						{
-							tabs2: selectedItem.type,
-						}
-					);
-
-					var data = {
-						redirect: assignmentsRedirect.toString(),
-					};
-
-					if (selectedItem.type == 'segments') {
-						data.addSegmentsEntryIds = selectedItem.value;
-					}
-					else if (selectedItem.type === 'users') {
-						data.addUserIds = selectedItem.value;
-					}
-					else {
-						data.addGroupIds = selectedItem.value;
-					}
-
-					Liferay.Util.postForm(form, {
-						data: data,
-						url: '<%= editRoleAssignmentsURL %>',
-					});
-				}
-			},
-			selectEventName: '<portlet:namespace />selectAssignees',
-			title:
-				'<liferay-ui:message arguments="<%= HtmlUtil.escape(role.getName()) %>" key="add-assignees-to-x" />',
-
-			<portlet:renderURL var="selectAssigneesURL" windowState="<%= LiferayWindowState.POP_UP.toString() %>">
-				<portlet:param name="mvcPath" value="/select_assignees.jsp" />
-				<portlet:param name="tabs2" value="<%= tabs2 %>" />
-				<portlet:param name="tabs3" value="available" />
-				<portlet:param name="redirect" value="<%= currentURL %>" />
-				<portlet:param name="roleId" value="<%= String.valueOf(roleId) %>" />
-				<portlet:param name="displayStyle" value="<%= displayStyle %>" />
-			</portlet:renderURL>
-
-			url: '<%= selectAssigneesURL %>',
-		});
-	};
-
-	<portlet:namespace />unsetRoleAssignments = function () {
-		var assigneeType = '<%= HtmlUtil.escapeJS(tabs2) %>';
-		var ids = Liferay.Util.listCheckedExcept(
-			form,
-			'<portlet:namespace />allRowIds'
-		);
-
-		var data = {
-			assignmentsRedirect: '<%= portletURL.toString() %>',
-		};
-
-		if (assigneeType == 'segments') {
-			data.removeSegmentsEntryIds = ids;
-		}
-		else if (assigneeType === 'users') {
-			data.removeUserIds = ids;
-		}
-		else {
-			data.removeGroupIds = ids;
-		}
-
-		Liferay.Util.postForm(form, {
-			data: data,
-			url: '<%= editRoleAssignmentsURL %>',
-		});
-	};
-
-	Liferay.componentReady('editRoleAssignmentsManagementToolbar').then(
-		(managementToolbar) => {
-			managementToolbar.on('creationButtonClicked', addAssignees);
-		}
-	);
-
-	var state = window.sessionStorage.getItem(
-		'<%= RolesAdminWebKeys.MODAL_SEGMENT_STATE %>'
-	);
-
-	if (state == 'open') {
-		window.sessionStorage.removeItem(
-			'<%= RolesAdminWebKeys.MODAL_SEGMENT_STATE %>'
-		);
-
-		addAssignees();
-	}
-</aui:script>
\ No newline at end of file
+<liferay-frontend:component
+	context="<%= additionalProps %>"
+	module="js/EditRoleAssignments"
+/>
```

Comment:
I'm very curious what other reviewers will say about this!

We discussed at length what's the best way to access a function from both JSP and a `PropsTransformer`. This way, both import it from the same file.

Reply:
I think this is a creative way to get access from both worlds using the existing infrastructure, kudos!

That being said I'm not super happy with the usage of `liferay-frontend:component`, because [it will output](https://github.com/liferay-frontend/liferay-portal/blob/master/modules/apps/frontend-taglib/frontend-taglib/src/main/java/com/liferay/frontend/taglib/servlet/taglib/ComponentTag.java#L196) something like: 

```
Liferay.component(
    'xvfx', 
    new jsEditRoleAssignments.default(...)
    {
        destroyOnNavigate: true,
        portletId: '...'
    }
);
```

So, basically we're just triggering a side-effect here via `new myFunction(data)` and then keeping that "instance" in our register of components.

I definitely wouldn't want this to become a pattern, although obviously we don't have a better one at this point. @izaera worked on the "launcher" concept which I think might be better suited for this since we simply want to run a JS snippet but we don't have an easy way to interface with it at the moment.

Reply:
If a full component creation is undesirable, I'd be ok with an inline script like what we did in `users-admin-web`: https://github.com/liferay/liferay-portal/blob/876bb19b1412a2c79089737c0f1b827dda6d1554/modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/view.jsp#L300-L311.

Reply:
Yeah, I think a simple script might be better in this case because of the semantics. It's still a bit weird because you have to do something like:

```html
<aui:script require='<%= npmResolvedPackageName +  "js/EditRoleAssignments" %>'>
jsEditRoleAssignments.default({
    ...
});
</aui:script>
```

Reply:
I actually think in this case we can just inline the whole EditRoleAssignments component. Then we can just import and call the `addAssignees` directly.

Reply:
@drewbrokke, would be worth noticing that `window['<portlet:namespace />openSelectUsersDialog']`... is something we're trying to avoid with some of these changes 🤞

Reply:
@jbalsas agreed, that pattern would not apply to this case.

Reply:
Hey @jbalsas, @drewbrokke thanks for the feedback.

Initially I wanted to do what is done [here](https://github.com/liferay-frontend/liferay-portal/blob/8b8e03937395d44aba2c99114824e685a6269794/modules/apps/roles/roles-admin-web/src/main/resources/META-INF/resources/js/EditRoleAssignments.js#L25-L37) inside the management toolbar via a callback, but it seems that we didn't want to add the `onComponentMount` callback back in [https://github.com/liferay-frontend/liferay-portal/pull/826](https://github.com/liferay-frontend/liferay-portal/pull/826) so this is what I came up with .

I knew there was [something](https://github.com/liferay-frontend/liferay-portal/pull/835#discussion_r583019982) I hadn't taken into account.

Reply:
> Initially I wanted to do what is done here inside the management toolbar via a callback, but it seems that we didn't want to add the onComponentMount callback back in #826 so this is what I came up with .

I think that wouldn't have been a good use case either. Code that qualified for an eventual `onComponentMount` should depend on that component being mounted to work. In this case, what we want to run has no connection whatsoever with the component. Similarly, you could run the code from the `propsTransformer` since you know it's going to execute, but that would make it a side-effect again.

Reply:
Yeah it's true that's not really related to the component. 

I'll try with 

```js
<aui:script require='<%= npmResolvedPackageName +  "js/EditRoleAssignments" %>'>
    jsEditRoleAssignments.default({
        ...
    });
</aui:script>
```

Reply:
The change in [https://github.com/liferay-frontend/liferay-portal/pull/835/commits/ff87e533e5ffe658bdf531af5cf8d5e68fb6cf2f](https://github.com/liferay-frontend/liferay-portal/pull/835/commits/ff87e533e5ffe658bdf531af5cf8d5e68fb6cf2f) works as expected but I'm not sure "this is the way".

I'll squash those extra commits before forwarding, but I thought it would easier to leave them on top during the review.